deployment/centos-package-x64/Dockerfile

```
@@ -13,12 +13,12 @@ RUN yum update -y \
  && yum install -y epel-release
 
 # Install build dependencies
-RUN yum install -y @buildsys-build rpmdevtools yum-plugins-core libcurl-devel fontconfig-devel freetype-devel openssl-devel glibc-devel libicu-devel git
+RUN yum install -y make rpmdevtools libcurl-devel fontconfig-devel freetype-devel openssl-devel glibc-devel libicu-devel git
 
 # Install recent NodeJS and Yarn
 RUN curl -fSsLo /etc/yum.repos.d/yarn.repo https://dl.yarnpkg.com/rpm/yarn.repo \
- && rpm -i https://rpm.nodesource.com/pub_10.x/el/7/x86_64/nodesource-release-el7-1.noarch.rpm \
- && yum install -y yarn
+ && curl -sL https://rpm.nodesource.com/setup_12.x | bash - \
```

Is it possible to avoid `curl foo | bash -` things? They're awfully insecure...

---

@JustAMan I would love to, but this is taken from their [installation documentation](https://github.com/nodesource/distributions/blob/master/README.md#installation-instructions-1) for "Enterprise Linux"

I did notice this line
> To compile and install native addons from npm you may also need to install build tools:
  yum install gcc-c++ make
  or: yum groupinstall 'Development Tools'

So if `gcc-c++` is required I can add that add that package as well.

---

> Is it possible to avoid `curl foo | bash -` things? They're awfully insecure...

I completely agree.  I'd flat-out refuse to run something like this.

---

```suggestion
 && rpm -i https://rpm.nodesource.com/pub_12.x/el/8/x86_64/nodesource-release-el8-1.noarch.rpm \
```
For node 12 only the version stuff in the url changed, no need for `curl | bash`.
Also newer Nodejs is now available as a Module Stream `dnf module install -y nodejs:12/common`.

---

That is a nice solution, especially the `dnf module install`, but if JF were to maintain a CentOS 7 build as well, then it would need to do the former
`rpm -i https://rpm.nodesource.com/pub_12.x/el/7/x86_64/nodesource-release-el7-1.noarch.rpm \`

I am still thinking of add `gcc-c++` back just in case it's required by JF web, since it was a part of the original `buildsys-build` package for CentOS 7.